elastic/kibana
# [Discover] Fix overwriting data view id when loading a saved search with ES|QL

## Summary

This PR prevents the propagation of the AppState index (which is the id of the data view in the URL), to the state of a loaded saved search containing a text based language. Since the id of the data view used in this case is driven from the text base language statement, there's no reason for mutating the saved search state

## Testing
* In Discover, create a SavedSearch with an ES|QL statement
* Reload the browser tab, note that the index value in the url changes
* Save the Saved Search again, with this PR the persisted data view id should not be index value in the URL, before this PR it was 

### Checklist
